@copilot 
19:56:44.402 Running build in Washington, D.C., USA (East) – iad1
19:56:44.403 Build machine configuration: 4 cores, 8 GB
19:56:44.526 Cloning github.com/gredice/gredice (Branch: codex/update-scheduled-planting-task-display, Commit: feec516)
19:56:45.855 Cloning completed: 1.329s
19:56:48.428 Restored build cache from previous deployment (ArMrMtRDcEPLafxJFU3ZwMQzPEDN)
19:56:48.808 Running "vercel build"
19:56:49.413 Vercel CLI 51.6.1
19:56:49.539 > Detected Turbo. Adjusting default settings...
19:56:50.151 > Detected ENABLE_EXPERIMENTAL_COREPACK=1 and "pnpm@10.33.0+sha512.10568bb4a6afb58c9eb3630da90cc9516417abebd3fabbe6739f0ae795728da1491e9db5a544c76ad8eb7570f5c4bb3d6c637b2cb41bfdcdb47fa823c8649319" in package.json
19:56:50.427 Running "install" command: `pnpm install`...
19:56:50.953 Scope: all 20 workspace projects
19:56:53.748 
19:56:53.923 Done in 3.4s using pnpm v10.33.0
19:56:53.949 Detected Next.js version: 16.2.4
19:56:53.950 Running "turbo run build"
19:56:54.331 
19:56:54.332    • Packages in scope: farm
19:56:54.332    • Running build in 1 packages
19:56:54.332    • Remote caching enabled
19:56:54.332 
19:56:54.470 farm:build: cache miss, executing 1f1fd0a30fcf4dce
19:56:54.785 farm:build: 
19:56:54.785 farm:build: > farm@0.0.0 build /vercel/path0/apps/farm
19:56:54.785 farm:build: > next build
19:56:54.785 farm:build: 
19:56:55.466 farm:build:   Applying modifyConfig from Vercel
19:56:55.489 farm:build: ▲ Next.js 16.2.4 (Turbopack)
19:56:55.490 farm:build: - Experiments (use with caution):
19:56:55.490 farm:build:   · optimizePackageImports
19:56:55.490 farm:build:   ✓ typedEnv
19:56:55.490 farm:build: 
19:56:55.515 farm:build:   Creating an optimized production build ...
19:57:08.784 farm:build: 
19:57:08.784 farm:build: > Build error occurred
19:57:08.787 farm:build: Error: Turbopack build failed with 2 errors:
19:57:08.787 farm:build: ./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx:1:1
19:57:08.788 farm:build: Module not found: Can't resolve '@gredice/js/plants'
19:57:08.788 farm:build: [31m[1m>[0m [90m1 |[0m [36mimport[0m { calculatePlantsPerField } [36mfrom[0m [32m'@gredice/js/plants'[0m;
19:57:08.788 farm:build:   [90m  |[0m [31m[1m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[0m
19:57:08.788 farm:build:   [90m2 |[0m [36mimport[0m [36mtype[0m {
19:57:08.788 farm:build:   [90m3 |[0m     [33mEntityStandardized[0m,
19:57:08.788 farm:build:   [90m4 |[0m     [33mRaisedBedFieldAssignableFarmUser[0m,
19:57:08.788 farm:build: 
19:57:08.788 farm:build: 
19:57:08.788 farm:build: 
19:57:08.788 farm:build: Import trace:
19:57:08.788 farm:build:   Server Component:
19:57:08.789 farm:build:     ./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
19:57:08.789 farm:build:     ./apps/farm/app/schedule/FarmSchedulePlantingsSectionContent.tsx
19:57:08.789 farm:build:     ./apps/farm/app/schedule/FarmScheduleDay.tsx
19:57:08.789 farm:build:     ./apps/farm/app/schedule/page.tsx
19:57:08.789 farm:build: 
19:57:08.789 farm:build: https://nextjs.org/docs/messages/module-not-found
19:57:08.789 farm:build: 
19:57:08.789 farm:build: 
19:57:08.789 farm:build: ./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx:8:1
19:57:08.789 farm:build: Module not found: Can't resolve '@gredice/ui/users'
19:57:08.789 farm:build:   [90m 6 |[0m [36mimport[0m { [33mLocalDateTime[0m } [36mfrom[0m [32m'@gredice/ui/LocalDateTime'[0m;
19:57:08.789 farm:build:   [90m 7 |[0m [36mimport[0m { [33mRaisedBedLabel[0m } [36mfrom[0m [32m'@gredice/ui/raisedBeds'[0m;
19:57:08.789 farm:build: [31m[1m>[0m [90m 8 |[0m [36mimport[0m { [33mUserAvatar[0m } [36mfrom[0m [32m'@gredice/ui/users'[0m;
19:57:08.789 farm:build:   [90m   |[0m [31m[1m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[0m
19:57:08.789 farm:build:   [90m 9 |[0m [36mimport[0m { [33mCheckbox[0m } [36mfrom[0m [32m'@signalco/ui-primitives/Checkbox'[0m;
19:57:08.789 farm:build:   [90m10 |[0m [36mimport[0m { [33mRow[0m } [36mfrom[0m [32m'@signalco/ui-primitives/Row'[0m;
19:57:08.789 farm:build:   [90m11 |[0m [36mimport[0m { [33mStack[0m } [36mfrom[0m [32m'@signalco/ui-primitives/Stack'[0m;
19:57:08.790 farm:build: 
19:57:08.790 farm:build: 
19:57:08.790 farm:build: 
19:57:08.790 farm:build: Import trace:
19:57:08.790 farm:build:   Server Component:
19:57:08.790 farm:build:     ./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
19:57:08.790 farm:build:     ./apps/farm/app/schedule/FarmSchedulePlantingsSectionContent.tsx
19:57:08.790 farm:build:     ./apps/farm/app/schedule/FarmScheduleDay.tsx
19:57:08.790 farm:build:     ./apps/farm/app/schedule/page.tsx
19:57:08.790 farm:build: 
19:57:08.790 farm:build: https://nextjs.org/docs/messages/module-not-found
19:57:08.790 farm:build: 
19:57:08.790 farm:build: 
19:57:08.790 farm:build:     at <unknown> (./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx:1:1)
19:57:08.790 farm:build:     at <unknown> (https://nextjs.org/docs/messages/module-not-found)
19:57:08.791 farm:build:     at <unknown> (./apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx:8:1)
19:57:08.791 farm:build:     at <unknown> (https://nextjs.org/docs/messages/module-not-found)
19:57:08.869 farm:build:  ELIFECYCLE  Command failed with exit code 1.
19:57:08.882  ERROR  farm#build: command (/vercel/path0/apps/farm) /vercel/path0/.vercel/cache/corepack/shim/pnpm run build exited (1)
19:57:08.882  WARNING  finished with warnings
19:57:08.883 Warning - the following environment variables are set on your Vercel project, but missing from "turbo.json". These variables WILL NOT be available to your application and may cause your build to fail. Learn more at https://turborepo.dev/docs/crafting-your-repository/using-environment-variables#platform-environment-variables
19:57:08.883 
19:57:08.883 [warn] farm#build
19:57:08.883 [warn]   - GREDICE_GOOGLE_MAPS_API_KEY 
19:57:08.883 [warn]   - GREDICE_GOOGLE_MAPS_URL_SIGNING_SECRET 
19:57:08.883 [warn]   - SLACK_BOT_TOKEN 
19:57:08.885 
19:57:08.885   Tasks:    0 successful, 1 total
19:57:08.885  Cached:    0 cached, 1 total
19:57:08.885    Time:    14.791s 
19:57:08.885 Summary:    /vercel/path0/.turbo/runs/3CXbnUxAojZt3Hd9UMMBYj17d37.json
19:57:08.885  Failed:    farm#build
19:57:08.885 
19:57:08.893  ERROR  run failed: command  exited (1)
19:57:08.907 Error: Command "turbo run build" exited with 1

--- a/apps/farm/app/flags.ts
+++ b/apps/farm/app/flags.ts
@@ -16,6 +16,4 @@ const hypertuneAdapter = createHypertuneAdapter<FlagValues, Context>({
     identify,
 });
 
-export const showUIFlag = flag(
-    hypertuneAdapter.declarations.showUI,
-);
+export const showUIFlag = flag(hypertuneAdapter.declarations.showUI);

--- a/apps/farm/app/schedule/FarmScheduleDay.tsx
+++ b/apps/farm/app/schedule/FarmScheduleDay.tsx
@@ -33,6 +33,7 @@ export function FarmScheduleDay({
                 <FarmSchedulePlantingsSectionContent
                     dayDataPromise={dayDataPromise}
                     plantSortsPromise={plantSortsPromise}
+                    userId={userId}
                 />
             </Suspense>
             <Suspense fallback={<FarmScheduleSectionSkeleton />}>

--- a/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
+++ b/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
@@ -1,6 +1,11 @@
-import type { EntityStandardized } from '@gredice/storage';
+import { calculatePlantsPerField } from '@gredice/js/plants';
+import type {
+    EntityStandardized,
+    RaisedBedFieldAssignableFarmUser,
+} from '@gredice/storage';
 import { LocalDateTime } from '@gredice/ui/LocalDateTime';
 import { RaisedBedLabel } from '@gredice/ui/raisedBeds';
+import { UserAvatar } from '@gredice/ui/users';
 import { Checkbox } from '@signalco/ui-primitives/Checkbox';
 import { Row } from '@signalco/ui-primitives/Row';
 import { Stack } from '@signalco/ui-primitives/Stack';
@@ -21,6 +26,8 @@ interface FarmSchedulePlantingsSectionProps {
     raisedBeds: FarmScheduleDayData['raisedBeds'];
     scheduledFields: FarmScheduleDayData['scheduledFields'];
     plantSorts: EntityStandardized[] | null | undefined;
+    userId: string;
+    assignedUserByFieldId: Map<number, RaisedBedFieldAssignableFarmUser>;
 }
 
 function buildFieldLabel(
@@ -30,13 +37,18 @@ function buildFieldLabel(
     const sort = field.plantSortId
         ? plantSortById.get(field.plantSortId)
         : null;
-    return `${field.positionIndex + 1} - sijanje${sort ? `: ${sort.information?.name ?? 'Nepoznato'}` : ''}`;
+    const { totalPlants } = calculatePlantsPerField(
+        sort?.information?.plant?.attributes?.seedingDistance,
+    );
+    return `${field.positionIndex + 1} - sijanje: ${totalPlants}${sort ? ` ${sort.information?.name ?? 'Nepoznato'}` : ''}`;
 }
 
 export function FarmSchedulePlantingsSection({
     raisedBeds,
     scheduledFields,
     plantSorts,
+    userId,
+    assignedUserByFieldId,
 }: FarmSchedulePlantingsSectionProps) {
     if (scheduledFields.length === 0) {
         return null;
@@ -116,11 +128,17 @@ export function FarmSchedulePlantingsSection({
                                     const approved = isFieldApproved(
                                         field.plantStatus,
                                     );
+                                    const canComplete =
+                                        !completed &&
+                                        (!field.assignedUserId ||
+                                            field.assignedUserId === userId);
+                                    const assignedUser =
+                                        assignedUserByFieldId.get(field.id);
 
                                     return (
                                         <div
                                             key={field.id}
-                                            className="rounded-lg border bg-white px-3 py-2"
+                                            className={`rounded-lg border bg-white px-3 py-2 ${!canComplete ? 'opacity-70' : ''}`}
                                         >
                                             <Row
                                                 spacing={1}
@@ -136,7 +154,7 @@ export function FarmSchedulePlantingsSection({
                                                             checked
                                                             disabled
                                                         />
-                                                    ) : (
+                                                    ) : canComplete ? (
                                                         <CompletePlantingModal
                                                             label={field.label}
                                                             raisedBedId={
@@ -146,6 +164,13 @@ export function FarmSchedulePlantingsSection({
                                                                 field.positionIndex
                                                             }
                                                         />
+                                                    ) : (
+                                                        <div title="Sijanje je dodijeljeno drugom korisniku.">
+                                                            <Checkbox
+                                                                className="size-5"
+                                                                disabled
+                                                            />
+                                                        </div>
                                                     )}
                                                     <Stack
                                                         spacing={0.5}
@@ -211,6 +236,23 @@ export function FarmSchedulePlantingsSection({
                                                         </Row>
                                                     </Stack>
                                                 </Row>
+                                                {assignedUser && (
+                                                    <div
+                                                        className="shrink-0"
+                                                        title={`Dodijeljeno: ${assignedUser.displayName ?? assignedUser.userName}`}
+                                                    >
+                                                        <UserAvatar
+                                                            avatarUrl={
+                                                                assignedUser.avatarUrl
+                                                            }
+                                                            displayName={
+                                                                assignedUser.displayName ??
+                                                                assignedUser.userName
+                                                            }
+                                                            className="size-7"
+                                                        />
+                                                    </div>
+                                                )}
                                             </Row>
                                         </div>
                                     );

--- a/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
+++ b/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
@@ -37,10 +37,7 @@ function buildFieldLabel(
     const sort = field.plantSortId
         ? plantSortById.get(field.plantSortId)
         : null;
-    if (!field.plantSortId) {
-        return `${field.positionIndex + 1} - sijanje: ?`;
-    }
-    if (!sort) {
+    if (!field.plantSortId || !sort) {
         return `${field.positionIndex + 1} - sijanje: ? Nepoznato`;
     }
 
@@ -48,8 +45,8 @@ function buildFieldLabel(
         sort.information?.plant?.attributes?.seedingDistance;
     const totalPlants = seedingDistance
         ? calculatePlantsPerField(seedingDistance).totalPlants
-        : '?';
-    return `${field.positionIndex + 1} - sijanje: ${totalPlants} ${sort.information?.name ?? 'Nepoznato'}`;
+        : null;
+    return `${field.positionIndex + 1} - sijanje: ${totalPlants ?? '?'} ${sort.information?.name ?? 'Nepoznato'}`;
 }
 
 export function FarmSchedulePlantingsSection({

--- a/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
+++ b/apps/farm/app/schedule/FarmSchedulePlantingsSection.tsx
@@ -5,7 +5,7 @@ import type {
 } from '@gredice/storage';
 import { LocalDateTime } from '@gredice/ui/LocalDateTime';
 import { RaisedBedLabel } from '@gredice/ui/raisedBeds';
-import { UserAvatar } from '@gredice/ui/users';
+import { UserAvatar } from '@gredice/ui/UserAvatar';
 import { Checkbox } from '@signalco/ui-primitives/Checkbox';
 import { Row } from '@signalco/ui-primitives/Row';
 import { Stack } from '@signalco/ui-primitives/Stack';
@@ -37,10 +37,19 @@ function buildFieldLabel(
     const sort = field.plantSortId
         ? plantSortById.get(field.plantSortId)
         : null;
-    const { totalPlants } = calculatePlantsPerField(
-        sort?.information?.plant?.attributes?.seedingDistance,
-    );
-    return `${field.positionIndex + 1} - sijanje: ${totalPlants}${sort ? ` ${sort.information?.name ?? 'Nepoznato'}` : ''}`;
+    if (!field.plantSortId) {
+        return `${field.positionIndex + 1} - sijanje: ?`;
+    }
+    if (!sort) {
+        return `${field.positionIndex + 1} - sijanje: ? Nepoznato`;
+    }
+
+    const seedingDistance =
+        sort.information?.plant?.attributes?.seedingDistance;
+    const totalPlants = seedingDistance
+        ? calculatePlantsPerField(seedingDistance).totalPlants
+        : '?';
+    return `${field.positionIndex + 1} - sijanje: ${totalPlants} ${sort.information?.name ?? 'Nepoznato'}`;
 }
 
 export function FarmSchedulePlantingsSection({
@@ -128,17 +137,19 @@ export function FarmSchedulePlantingsSection({
                                     const approved = isFieldApproved(
                                         field.plantStatus,
                                     );
-                                    const canComplete =
+                                    const lockedByAssignment =
                                         !completed &&
-                                        (!field.assignedUserId ||
-                                            field.assignedUserId === userId);
+                                        !!field.assignedUserId &&
+                                        field.assignedUserId !== userId;
+                                    const canComplete =
+                                        !completed && !lockedByAssignment;
                                     const assignedUser =
                                         assignedUserByFieldId.get(field.id);
 
                                     return (
                                         <div
                                             key={field.id}
-                                            className={`rounded-lg border bg-white px-3 py-2 ${!canComplete ? 'opacity-70' : ''}`}
+                                            className={`rounded-lg border bg-white px-3 py-2 ${lockedByAssignment ? 'opacity-70' : ''}`}
                                         >
                                             <Row
                                                 spacing={1}

--- a/apps/farm/app/schedule/FarmSchedulePlantingsSectionContent.tsx
+++ b/apps/farm/app/schedule/FarmSchedulePlantingsSectionContent.tsx
@@ -1,3 +1,7 @@
+import {
+    getAssignableFarmUsersByRaisedBedFieldIds,
+    type RaisedBedFieldAssignableFarmUser,
+} from '@gredice/storage';
 import { FarmSchedulePlantingsSection } from './FarmSchedulePlantingsSection';
 import type { FarmScheduleDayData } from './scheduleData';
 
@@ -6,11 +10,13 @@ interface FarmSchedulePlantingsSectionContentProps {
     plantSortsPromise: ReturnType<
         typeof import('./scheduleData').getFarmSchedulePlantSorts
     >;
+    userId: string;
 }
 
 export async function FarmSchedulePlantingsSectionContent({
     dayDataPromise,
     plantSortsPromise,
+    userId,
 }: FarmSchedulePlantingsSectionContentProps) {
     const { raisedBeds, scheduledFields } = await dayDataPromise;
 
@@ -18,13 +24,36 @@ export async function FarmSchedulePlantingsSectionContent({
         return null;
     }
 
-    const plantSorts = await plantSortsPromise;
+    const [plantSorts, assignableFarmUsersByRaisedBedFieldId] =
+        await Promise.all([
+            plantSortsPromise,
+            getAssignableFarmUsersByRaisedBedFieldIds(
+                scheduledFields.map((field) => field.id),
+            ),
+        ]);
+    const assignedUserByFieldId = new Map<
+        number,
+        RaisedBedFieldAssignableFarmUser
+    >();
+    for (const field of scheduledFields) {
+        if (!field.assignedUserId) {
+            continue;
+        }
+        const assignedUser = (
+            assignableFarmUsersByRaisedBedFieldId[field.id] ?? []
+        ).find((user) => user.id === field.assignedUserId);
+        if (assignedUser) {
+            assignedUserByFieldId.set(field.id, assignedUser);
+        }
+    }
 
     return (
         <FarmSchedulePlantingsSection
             raisedBeds={raisedBeds}
             scheduledFields={scheduledFields}
             plantSorts={plantSorts}
+            userId={userId}
+            assignedUserByFieldId={assignedUserByFieldId}
         />
     );
 }

--- a/apps/farm/app/schedule/actions.ts
+++ b/apps/farm/app/schedule/actions.ts
@@ -45,6 +45,10 @@ async function assertFarmerCanCompletePlanting(
         throw new Error('Nemaš dozvolu za označavanje ovog sijanja.');
     }
 
+    if (field.assignedUserId && field.assignedUserId !== userId) {
+        throw new Error('Ovo sijanje je dodijeljeno drugom korisniku.');
+    }
+
     return field;
 }
 

--- a/apps/farm/package.json
+++ b/apps/farm/package.json
@@ -37,6 +37,7 @@
     "devDependencies": {
         "@biomejs/biome": "2.4.12",
         "@gredice/client": "workspace:*",
+        "@gredice/js": "workspace:*",
         "@gredice/storage": "workspace:*",
         "@gredice/ui": "workspace:*",
         "@mdxeditor/editor": "3.54.1",

